### PR TITLE
Switch from Hugo to MkDocs with Material theme

### PR DESCRIPTION
**Pull request by rhagenson**
This PR mirrors the work on ponylang/pony-patterns#47

I will squash all the changes once this is ready to merge, but working in incremental commits so progress is easily tracked.

I know I still need to do the following:
- [x] Remove the YAML headers on individual pages (extracting the title into a level 1 header)
- [x] Resolve internal links
- [x] Enable Pony syntax highlighting
- [ ] Add the features that were added after the original Patterns PR (e.g., htmlproofer)
- [x] Enable the Insiders features
- [x] Check for formatting errors
- [x] Relocate file in static/

### CODE REVIEW

**Comment by rhagenson** [comment]
All necessary changes should be made. I will be submitting the changes to re-enable htmlproofer after this is accepted as the document edit links cause htmproofer to always fail until these changes hit main.

File: docs/expressions/arithmetic.md [hunk 1]
```diff
@@ -184,21 +176,21 @@ This allows more aggressive optimizations and for faster execution, but only yie
...
-`%~`     | rem_unsafe()
-`%%~`    | mod_unsafe()
-`-~`     | neg_unsafe()
-`<~`     | lt_unsafe()
-`>~`     | gt_unsafe()
-`<=~`    | le_unsafe()
-`>=~`    | ge_unsafe()
-`=~`     | eq_unsafe()
-`!=~`    | ne_unsafe()
+| Operator | Method       |
```
Comment (SeanTAllen) [inline docs/expressions/arithmetic.md:179]: this looks a bit more irritating to maintain.

File: netlify.toml [hunk 2]
```diff
@@ -7,45 +7,8 @@ ID = "Your_Site_ID"
...
-[context.production.environment]
-  HUGO_VERSION = "0.52"
-
-[context.deploy-preview.environment]
-  HUGO_VERSION = "0.52"
-
-[context.branch-deploy.environment]
-  HUGO_VERSION = "0.52"
-
-[[redirects]]
```
Comment (SeanTAllen) [inline netlify.toml:23]: these redirects should be kept and updated as appropriate. There's old links in the wild that will break if these aren't maintained.

File: docs/expressions/arithmetic.md (hunk 1 above)
Comment (rhagenson) [reply docs/expressions/arithmetic.md:179]: Are you talking about the additional `|` before and after all tables? I did not realize that was happening as it is part of my VS Code setup to run GFM table formatting on save.

No real opinion on reversing this as it was not intentionally done in the first place,

File: netlify.toml (hunk 2 above)
Comment (rhagenson) [reply netlify.toml:23]: Redirects have been put back in.

**Review by SeanTAllen — APPROVED** [review]

File: docs/expressions/arithmetic.md (hunk 1 above)
Comment (SeanTAllen) [reply docs/expressions/arithmetic.md:179]: Ah, I assumed it was required. Yeah, not having to have is nice.

File: netlify.toml (hunk 2 above)
Comment (SeanTAllen) [reply netlify.toml:23]: Awesome.

**Pull request merged**